### ISSUE TITLE
Bump deprecated apiversions

### ISSUE THREAD

**Issue by jpkrohling** [open-telemetry/opentelemetry-operator#188]
When installing the operator in new Kubernetes versions, the following warnings are shown:

```
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition

Warning: rbac.authorization.k8s.io/v1beta1 ClusterRole is deprecated in v1.17+, unavailable in v1.22+; use rbac.authorization.k8s.io/v1 ClusterRole

Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration

Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
```

We should bump them before 1.22.

**Comment by bhiravabhatla** [comment]
Picking this up

**Comment by bhiravabhatla** [comment]
There is an open issue for preserveUnknownFields=false with crdVersions=v1

https://github.com/kubernetes-sigs/controller-tools/issues/476

Let me know how would you want to handle this @jpkrohling

**Comment by jpkrohling** [comment]
Are you seeing this with recent Kubernetes versions as well?

**Comment by bhiravabhatla** [comment]
This is issue with controller-gen, when it generates manifests -

```
$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 ```
If we mention V1 as CRD version, it omits adding preserveUnknownFields=false to the base CRDs.

**Comment by jpkrohling** [comment]
Right, but if you apply the generted CRD in a recent Kubernetes cluster, does it complain?

**Comment by bhiravabhatla** [comment]
Actually, I faced that issue in an earlier project. Currently when i install CRDs I am seeing something else altogether - Looking like something to do with kustomize:

```
Error: no matches for OriginalId apiextensions.k8s.io_v1beta1_CustomResourceDefinition|~X|opentelemetrycollectors.opentelemetry.io; no matches for CurrentId apiextensions.k8s.io_v1beta1_CustomResourceDefinition|~X|opentelemetrycollectors.opentelemetry.io; failed to find unique target for patch apiextensions.k8s.io_v1beta1_CustomResourceDefinition|opentelemetrycollectors.opentelemetry.io
```

**Comment by bhiravabhatla** [comment]
Ok I figured it out - we need to modify patches mentioned for webhook

**Comment by bhiravabhatla** [comment]
Also @jpkrohling  - controller-gen version 0.4.0 defaults apiversion to v1. Shall we upgrade the version of controller-gen.

https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.4.0

**Comment by jpkrohling** [comment]
Sure, sounds good.

**Comment by bhiravabhatla** [comment]
Regarding conversionReviewVersions - This was introduced as mandatory field in v1 apiVersion for webhook spec ( Ref - https://github.com/kubernetes/kubernetes/pull/79604)

I am adding 

```
      conversionReviewVersions:
        - v1
        - v1beta1
```
But I also see that in k8s versions (1.17+) only v1 would be supported. What is the minimum supported version of k8s for our operator.

Ref:
https://github.com/kubernetes/kubernetes/issues/82023

**Comment by jpkrohling** [comment]
We have not defined yet, but I think a good policy is what Kubernetes itself supports: up to N-2.

**Comment by jpkrohling** [comment]
Closed by #220.
